File: encapsulation/enc.py
class Student:
    def __init__(self, name, age):
        self.name = name
        self.age = age

    def set_name(self, name):
        self.name = name

    def get_name(self):
        return self.name

    def set_age(self, age):
        self.age = age

    def get_age(self):
        return self.age


# Main code
if __name__ == "__main__":
    # Create an instance of the Student class by providing name and age
    std = Student("Daniel", 20)

    # Using the setter method to change the name
    std.set_name("David")
    std.set_age(25)

    # Retrieve and print the updated name
    print("This student's name is", std.get_name())
    print("This student's age is", std.get_age())

File: encapsulation/enc.java
package encapsulation;

public class enc {
    public static void main(String[] args) {

        Student student = new Student();

        // Set student information using setter methods
        student.setName("John Doe");
        student.setAge(25);

        // Retrieve and display student information using getter methods
        System.out.println("Student Name: " + student.getName());
        System.out.println("Student Age: " + student.getAge());

    }
}

class Student {
    private String name;
    private int age;

    // Getter and Setter methods for encapsulation
    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public int getAge() {
        return age;
    }

    public void setAge(int age) {
        this.age = age;
    }
}